### ISSUE TITLE
calculate-ttl doesn't work with Instants

### ISSUE THREAD

**Issue by lukaszkorecki** [cognitect-labs/aws-api#160]
Hi there! I've run into this issue when implementing two credential providers (Fargate+ECS and SSO profiles):

`calculate-ttl` will fail when an instant is passed as expiration time. This can happen when obtaining credentials via a call to https://docs.aws.amazon.com/singlesignon/latest/PortalAPIReference/API_GetRoleCredentials.html

In this case - expiration time is returned as a unix timestamp, which can be converted via `Instant/ofEpochMillis`. That however, requires further conversion to `Date`, because `calculate-ttl` attempts to call `.toInstant` on an instance of `Instant`.

The affected code can be found here: https://github.com/cognitect-labs/aws-api/blob/c4ca969975270ebbb5de18c7ad25b6fd391bf061/src/cognitect/aws/credentials.clj#L253-L255

The reason behind this issue is a bit odd - `inst?` returns true for both `java.util.Date` and `java.time.Instant`.

I have created a patch: https://github.com/cognitect-labs/aws-api/compare/master...lukaszkorecki:master

### Dependencies:

```

                 [com.cognitect.aws/api        "0.8.484"]
                 [com.cognitect.aws/endpoints "1.1.11.914"]
                 [com.cognitect.aws/s3       "810.2.801.0"]
```

**Comment by dchelimsky** [comment]
`calculate-ttl` accepts " java.util.Date, or a string parsable by java.time.Instant/parse". One solution would be to change this to accept Instant as well, but couldn't you parse a Date from the timestamp instead of parsing an instant with Instant/ofEpochMillis?

**Comment by lukaszkorecki** [comment]
Definitely, that's the approach I'm taking now - it's just that the code is misleading a bit, since it looks like it could just work with `Instant` - as that's what is being used internally.
